doc/development/legacy_plugins.rst

```diff
@@ -0,0 +1,532 @@
+
+Defining the ``__init__`` method of a custom device is not necessary; by default,
+the :class:`~.QubitDevice` initialization will be called, where the user can pass the
+following arguments:
+
+* ``wires`` (*int* or *Iterable[Number, str]*): The number of subsystems represented by the device,
+  or an iterable that contains unique labels for the subsystems as numbers (e.g., ``[-1, 0, 2]``)
+  and/or strings (``['auxiliary', 'q1', 'q2']``).
+
+* ``shots=1000`` (*None*, *int* or *List[int]*): number of circuit
+  evaluations/random samples used to estimate probabilities, expectation
+  values, variances  of observables in non-analytic mode. If ``shots=None``, the device
+  calculates probability, expectation values, and variances analytically. If `shots` is an
+  integer, it specifies the number of samples to estimate these quantities. If a
+  list of integers is passed, the circuit evaluations are batched over the list
+  of shots.
+
+To add your own device arguments, or to override any of the above defaults, simply
+overwrite the ``__init__`` method. For example, here is a device where the number
+of wires is fixed to ``24``, that cannot be used in analytic mode, and that can accept a dictionary
+of low-level hardware control options:
+
+.. code-block:: python3
+
+    class CustomDevice(QubitDevice):
+        name = 'My custom device'
+        short_name = 'example.mydevice'
+        pennylane_requires = '0.1.0'
+        version = '0.0.1'
+        author = 'Ada Lovelace'
+
+        operations = {"PauliX", "RX", "CNOT"}
+        observables = {"PauliZ", "PauliX", "PauliY"}
+
+        def __init__(self, shots=1024, hardware_options=None):
+            super().__init__(wires=24, shots=shots)
+            self.hardware_options = hardware_options or hardware_defaults
+
+Note that we have also overridden the default shot number.
+
+The user can now pass any of these arguments to the PennyLane device loader:
+
+>>> dev = qml.device("example.mydevice", hardware_options={"t2": 0.1})
+>>> dev.hardware_options
+{"t2": 0.1}
+
+
+Device execution
+----------------
+
+Once all of the class attributes are defined, it is necessary to define some required class
+methods to allow PennyLane to apply operations and measure observables on your device.
+
+To execute operations on the device, the following methods **must** be defined:
+
+.. currentmodule:: pennylane.QubitDevice
+
+.. autosummary::
+
+    apply
+
+If the device is a statevector simulator (it can perform analytic computations when ``shots=None``)
+then it **must** also overwrite:
+
+.. autosummary::
+
+    analytic_probability
+
+The :class:`~.QubitDevice` class
+provides the following convenience methods that may be used by the plugin:
+
+.. autosummary::
+
+    active_wires
+    marginal_prob
+
+In addition, if your qubit device generates its own computational basis samples for measured wires
+after execution, you need to overwrite the following method:
+
+.. autosummary::
+
+    generate_samples
```

Comment:
```suggestion
.. currentmodule:: pennylane.devices

.. autosummary::

    ~QubitDevice.apply

If the device is a statevector simulator (it can perform analytic computations when ``shots=None``)
then it **must** also overwrite:

.. autosummary::

    ~QubitDevice.analytic_probability

The :class:`~pennylane.devices.QubitDevice` class
provides the following convenience methods that may be used by the plugin:

.. autosummary::

    ~QubitDevice.active_wires
    ~QubitDevice.marginal_prob

In addition, if your qubit device generates its own computational basis samples for measured wires
after execution, you need to overwrite the following method:

.. autosummary::

    ~QubitDevice.generate_samples
```